tests/query/test_query_client_settings.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+
+import ydb
+
+from datetime import date, datetime, timedelta, timezone
+
+
+@pytest.fixture
+def settings_on():
+    settings = (
+        ydb.QueryClientSettings()
+        .with_native_date_in_result_sets(True)
+        .with_native_datetime_in_result_sets(True)
+        .with_native_timestamp_in_result_sets(True)
+        .with_native_interval_in_result_sets(True)
+        .with_native_json_in_result_sets(True)
+    )
+    return settings
+
+
+@pytest.fixture
+def settings_off():
+    settings = (
+        ydb.QueryClientSettings()
+        .with_native_date_in_result_sets(False)
+        .with_native_datetime_in_result_sets(False)
+        .with_native_timestamp_in_result_sets(False)
+        .with_native_interval_in_result_sets(False)
+        .with_native_json_in_result_sets(False)
+    )
+    return settings
+
+
+test_td = timedelta(microseconds=-100)
+test_now = datetime.utcnow()
+test_today = test_now.date()
+test_dt_today = datetime.today()
+tz4h = timezone(timedelta(hours=4))
+
+
+params = pytest.mark.parametrize(
+    "value,ydb_type,casted_result,uncasted_type",
```

Comment:
casted_result and uncasted_type are difficult to underdstand.

what about: raw result and native_result?
and always compare values